From f8d98f2e1b2b5f680624cde2b61c3486166b1ff8 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 31 Dec 2013 12:50:22 +0800
Subject: [PATCH 248/280] ENGR00293635 Report sequence change, FW v3.1.1
 workaround

VPU lib v5.4.22

- report sequence parameters change
Bit 20 of decodingSuccess is set to 1 when sequence parameters
have been changed.
VPU quits picture decoding with -1 indexFrameDecoded and a valid
indexFrameDisplay.

- workaround for FW v3.1.1 bug: BWB blocking issue (ENGR00293425)
Diable BWB for all decoders before FW bug is fixed.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_lib.c | 7 ++++++-
 vpu/vpu_lib.h | 4 ++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index aed65da..c925830 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1,7 +1,7 @@
 /*
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  *
- * Copyright (C) 2004-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2004-2014 Freescale Semiconductor, Inc.
  */
 
 /* The following programs are the sole property of Freescale Semiconductor Inc.,
@@ -2750,6 +2750,9 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 			val |= (pDecInfo->tiledLinearEnable << 11 | 0x03 << 9);
 		val |= 1 << 12;
 
+		/* workaround for BWB blocking issue (ENGR00293425), FW v3.1.1 */
+		val &= ~(1 << 12);
+
 		// workaround for BWB blocking issue (ENGR00231107)
 		if (pDecInfo->openParam.bitstreamFormat == STD_VC1)
 			val &= ~(1 << 12);
@@ -4284,6 +4287,8 @@ RetCode vpu_DecGetOutputInfo(DecHandle handle, DecOutputInfo * info)
 			info->decodingSuccess |= 0x10;
 			VpuWriteReg(BIT_RUN_INDEX, pCodecInst->instIndex);
 		}
+
+		info->decodingSuccess |= val & (1 << 20);
 	}
 
 	if (pCodecInst->codecMode == AVC_DEC) {
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index 5ef1bca..92489c7 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -1,7 +1,7 @@
 /*
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  *
- * Copyright (C) 2004-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2004-2014 Freescale Semiconductor, Inc.
  */
 
 /* The following programs are the sole property of Freescale Semiconductor Inc.,
@@ -683,7 +683,7 @@ static const SocInfo soc_info[] = {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 21)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 22)
 
 extern unsigned int system_rev;
 
-- 
1.8.0

